quai/filters/api.go

```diff
@@ -335,26 +335,43 @@ func (api *PublicFilterAPI) Accesses(ctx context.Context, addr common.Address) (
 		}()
 		api.activeSubscriptions += 1
 		headers := make(chan *types.WorkObject)
-		headersSub := api.events.SubscribeNewHeads(headers)
+		headersSub := api.events.SubscribeChainHeadEvent(headers)
 		unlocks := make(chan core.UnlocksEvent)
 		unlocksSub := api.events.SubscribeUnlocks(unlocks)
 		for {
 			select {
 			case h := <-headers:
+
 				// Marshal the header data
 				hash := h.Hash()
 				nodeLocation := api.backend.NodeLocation()
-				nodeCtx := nodeLocation.Context()
-				if block, err := api.backend.GetBlock(hash, h.NumberU64(nodeCtx)); err != nil {
-					for _, tx := range block.Transactions() {
-						// Check for external accesses
-						if tx.To() == &addr || tx.From(nodeLocation) == &addr {
+				for _, tx := range h.Transactions() {
+					// Check for external accesses
+					if tx.Type() == types.QuaiTxType {
```

Comment:
Case statement would be nicest here for readability

Reply:
yes, done